atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java

```diff
@@ -81,10 +90,59 @@
     private final SqlConnectionSupplier connections;
 
     public static DbKvs create(DbKeyValueServiceConfig config) {
-        return new DbKvs(config, config.tableFactorySupplier().get(), config.sqlConnectionSupplier());
+        Preconditions.checkArgument(config.connection().isPresent(),
```

Comment:
This is done to support internal use cases that require constructing essentially broken config, but then not using that config.  I already added this jank-y-ness to https://github.com/palantir/atlasdb/issues/504.